build.assets/tooling/cmd/build-apt-repos/apt_repo_tool.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"io/fs"
+	"log"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+type AptRepoTool struct {
+	config       *Config
+	aptly        *Aptly
+	s3Manager    *s3manager
+	supportedOSs *map[string][]string
+}
+
+// Instantiates a new apt repo tool instance and performs any required setup/config.
+func NewAptRepoTool(config *Config, supportedOSs *map[string][]string) (*AptRepoTool, error) {
+	art := &AptRepoTool{
+		config:       config,
+		s3Manager:    NewS3Manager(*config.bucketName),
+		supportedOSs: supportedOSs,
+	}
+
+	aptly, err := NewAptly()
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to create a new aptly instance")
+	}
+	art.aptly = aptly
+
+	return art, nil
+}
+
+// Runs the tool, performing all of the business logic outside of argument parsing.
+func (art *AptRepoTool) Run() error {
+	start := time.Now()
+	logrus.Infoln("Starting APT repo build process...")
+
+	err := art.s3Manager.DownloadExistingRepo(*art.config.localBucketPath)
+	if err != nil {
+		return trace.Wrap(err, "failed to sync existing repo from S3 bucket")
+	}
+
+	repos, err := art.createRepos()
+	if err != nil {
+		return trace.Wrap(err, "failed to create repos")
+	}
+
+	err = art.importNewDebs(repos)
+	if err != nil {
+		return trace.Wrap(err, "failed to import new debs")
+	}
+
+	err = art.publishRepos(repos)
+	if err != nil {
+		return trace.Wrap(err, "failed to publish repos")
+	}
+
+	aptlyRootDir, err := art.aptly.GetRootDir()
+	if err != nil {
+		return trace.Wrap(err, "failed to get Aptly root directory")
+	}
+
+	err = art.s3Manager.UploadBuiltRepo(filepath.Join(aptlyRootDir, "public"))
+	if err != nil {
+		return trace.Wrap(err, "failed to sync changes to S3 bucket")
+	}
+
+	execution_duration := time.Since(start).Round(time.Millisecond)
+	logrus.Infof("APT repo build process completed in %s\n", execution_duration)
```

Comment:
Remove extra variable and `\n`.
```suggestion
	logrus.Infof("APT repo build process completed in %s",
	    time.Since(start).Round(time.Millisecond))
```

Reply:
Var removed. Does `sprintf` or `logrus.Infof` in Go differ from `sprintf` in C in that one of them appends new lines?